src/App.tsx

```diff
@@ -1,27 +1,75 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+const enum ListFilterType {
+  All,
+  First5,
+  Red,
+  None,
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+function getFiltredList(filrer: ListFilterType) {
+  switch (filrer) {
+    case ListFilterType.All:
+      return getAll();
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+    case ListFilterType.First5:
+      return get5First();
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+    case ListFilterType.Red:
+      return getRedGoods();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+    default:
+      return null;
+  }
+}
 
-    <GoodsList goods={[]} />
-  </div>
-);
+export const App: React.FC = () => {
+  const [goodList, setGoodList] = useState<Good[]>([]);
+  const [filter, setFilter] = useState<ListFilterType>(ListFilterType.None);
+
+  useEffect(() => {
+    getFiltredList(filter)?.then(products => setGoodList(products));
+  }, [filter]);
```

Comment:
try to remove `filter` state variable and use `useCallback` hook to load goods on click